addons/sale_loyalty/models/sale_order.py

```diff
@@ -652,7 +652,7 @@ def _get_claimable_rewards(self, forced_coupons=None):
         Coupons that can not claim any reward are not contained in the result.
         """
         self.ensure_one()
-        all_coupons = forced_coupons or (self.coupon_point_ids.coupon_id | self.order_line.coupon_id | self.applied_coupon_ids)
+        all_coupons = forced_coupons or (self.coupon_point_ids.coupon_id | self.applied_coupon_ids)
```

Comment:
IMHO, all_coupons shouldn't change. Whether they were applied on a SO or a SOL shouldn't be the question.
So, instead, add a condition when checking each coupon, to discard this case for discount only and not mark the reward as being claimable?

Reply:
This was the first thought I had in fixing this :sweat_smile:. But the condition will check if the coupon is present in self.order_line.coupon_id and if so it will be discarded so we will be eventually discarding every coupon in self.order_line.coupon_id so we can just remove it from all_coupons. Since self.order_line.coupon_id will have all the discount that is already applied in the SO correct me if I am wrong

Reply:
Hmmm, no, you have to only discard the ones that are discounts. AFAIK, claiming free products or so on could still be done multiple time (we might rework that kind of thing, but in master, not in stable), but it doesn't make sense for discounts, indeed